### ISSUE TITLE
Config Valid API Key has JS error Cannot read property 'length' of undefined "stores".

### ISSUE THREAD

**Issue by LiamKarlMitchell** [mailchimp/mc-magento2#977]
Please wait... and spins forever. There is a JS Error when attempting to configure valid API Key.
Valid response does not have stores array?

### Preconditions

Preconditions:
1. Magento CE 2.3.2 is installed without sample data.
2. Installed mc-magento 102.3.37
3. I am using a Mailchimp login that was invited to the project on Mailchimp as an Admin.

### Steps to reproduce

Steps to reproduce:
1. Attempting to add API Key to plugin config in Admin area.
2. Login to Mailchimp as expected.
3. After pasting in valid API Key and clicking off the API Key input element, the Ajax Loader says 

### Actual and Expected result

Actual result:
Has a javascript error because the response has no array named stores.

Expected result:
Should Work and bring back list of details etc to continue configuring the Mailchimp Store to Magento 2.

The next select list should show me a list of stores.

### Additional information

Javascript Error:
```
configapikey.js:108 Uncaught TypeError: Cannot read property 'length' of undefined
    at Object.<anonymous> (configapikey.js:108)
    at fire (jquery.js:3232)
    at Object.fireWith [as resolveWith] (jquery.js:3362)
    at done (jquery.js:9840)
    at XMLHttpRequest.callback (jquery.js:10311)
```

Network Response is:
{"valid":1}

![image](https://user-images.githubusercontent.com/1968109/76485479-80a3d200-6481-11ea-9095-a0d83b27ca37.png)

Suggested Improvements:
Handle case when stores array is not present or empty length and tell user what to do to fix?
Update documentation to say how to make the stores?

Was trying to follow documentation and also video guide:
https://mailchimp.com/help/connect-or-disconnect-mailchimp-for-magento-2/
https://www.youtube.com/watch?v=_6mM7Hew5aw

![image](https://user-images.githubusercontent.com/1968109/76485456-7aadf100-6481-11ea-9e0e-d0a1aeba12d0.png)

![image](https://user-images.githubusercontent.com/1968109/76485533-a4ffae80-6481-11ea-8048-440a5e8b1815.png)

Duplicate of:
https://github.com/mailchimp/mc-magento2/issues/950
Which was closed for not following the issue formatting.

What else do I have to do to configure this?
Is there another step in the actual Mailgun to add a Store?

**Comment by LiamKarlMitchell** [comment]
Sorry, further searching I came across more duplicates.
https://github.com/mailchimp/mc-magento2/issues/944
https://github.com/mailchimp/mc-magento2/issues/940

@Santiagoebizmarts  said the problem is no audience on the Mailchimp Account.
To create one and this will not be a problem anymore?

The JS code should tell the user this helpful information.
```
if (typeof data.stores === "undefined") {
  alert("Please first create an Audience in your Mailchimp Account.");
}
```
etc..

Only problem still remains because this Mailchimp account does indeed have an Audience.

![image](https://user-images.githubusercontent.com/1968109/76485937-c57c3880-6482-11ea-8c6a-0968ce6892e4.png)

Named: "Subscribers" created 2014.
Maybe there is some kind of API limitation where it won't get old ones?

I created an additional new Audience and tried logging in and adding the key again but no luck.

**Comment by LiamKarlMitchell** [comment]
I found under the top right menu on Mailchimp site.

Connected Sites.
It is already connected to the current Magento 1 website, I am developing a Magento 2 site to replace it due to the EOL.

When I click to add Magento 2  I am taken to the marketplace Mailchimp extension page, which is where I got the extension from yesterday.

I see that the Current Version on marketplace is 102.3.37 and the latest release on this github is the same.
